packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -231,7 +198,298 @@ const reason = "User declined to authenticate."
+- `id`: The wallet's internal identifier for the user. If no other identifier is used, simply the user's flow account address can be used here.
+- `identity`: Information about the identity of the user.
+- `provider`: Information about the wallet.
+- `data`: Additional information used with a service of type `open-id`.
+
+See also:
+- [authn](https://github.com/onflow/flow-js-sdk/blob/master/packages/fcl/src/current-user/normalize/authn.js)
+- [authz](https://github.com/onflow/flow-js-sdk/blob/master/packages/fcl/src/current-user/normalize/authz.js)
+- [user-signature](https://github.com/onflow/flow-js-sdk/blob/master/packages/fcl/src/current-user/normalize/user-signature.js)
+- [pre-authz](https://github.com/onflow/flow-js-sdk/blob/master/packages/fcl/src/current-user/normalize/pre-authz.js)
+- [open-id](https://github.com/onflow/flow-js-sdk/blob/master/packages/fcl/src/current-user/normalize/open-id.js)
+- [back-channel-rpc](https://github.com/onflow/flow-js-sdk/blob/master/packages/fcl/src/current-user/normalize/back-channel-rpc.js)
+
+### `Identity`
+This object is used to define the identity of the user.
+
+```typescript
+interface Identity extends ObjectBase {
+  f_type: 'Identity'
+  address: string
+  keyId?: number
+}
+```
+
+The meaning of the fields is as follows.
+- `address`: The flow account address of the user.
+- `keyId`: The user's key id.
+
+### `ServiceProvider`
+This object is used to communicate information about a wallet.
+
+```typescript
+interface ServiceProvider extends ObjectBase {
+  f_type: 'ServiceProvider'
+  address: string
+  name?: string
+  description?: string
+  icon?: string
+  website?: string
+  supportUrl?: string
+  supportEmail?: string
+}
+```
+
+The meaning of the fields is as follows.
+- `address`: A flow account address owned by the wallet. It is unspecified what this will be used for.
+- `name`: The name of the wallet.
+- `description`: A short description for the wallet
+- `icon`: An image URL for the wallet's icon.
+- `website`: The wallet's website.
+- `supportUrl`: A URL the user can use to get support with the wallet.
+- `supportEmail`: An e-mail address the user can use to get support with the wallet.
+
+### `AuthnResponse`
+This object is used to inform FCL about the services a wallet provides.
+
+```typescript
+interface AuthnResponse extends ObjectBase {
+  f_type: 'AuthnResponse'
+  addr: string
+  services: Service[]
+}
+```
+
+The meaning of the fields is as follows.
+- `addr`: The flow account address of the user.
+- `services`: The list of services provided by the wallet.
+
+### `Signable`
+
+```typescript
+interface FlowObjectSignable extends ObjectBase<'1.0.1'> {
+  f_type: 'Signable'
+  message: string
+  addr: string
+  keyId: number
+  roles: {
+    authorizer: boolean
+    param: boolean
+    payer: boolean
+    proposer: boolean
+  }
+  cadence: string
+  args: {
+    type: string
+    value: string
+  }[]
+  data: unknown
+  interaction: unknown
+  voucher: unknown
```

Comment:
These should be expanded later. Or not. Should the wallet care about any fields here other than the `voucher`?